### DISCUSSION TITLE
Making changes in degradation function:

### DISCUSSION THREAD

**Discussion by Aayush-T** [idaholab/moose#28546] [Q&A Modules: Phase field]
So, I am working on a crystal plasticity phase field problem. I based my code on input file: combined/test/tests/phase_field_fracture/crack2d_aniso.i
Here, and in many other files, a very simple degradation function was defined as:
```
[./degradation]
    type = DerivativeParsedMaterial
    property_name = degradation
    coupled_variables = 'c'
    expression = '(1.0-c)^2*(1.0 - eta) + eta'
    constant_names       = 'eta'
    constant_expressions = '1.0e-6'
    derivative_order = 2
  [../]
```
I wanted to modify it as:
```
[./degradation]
    type = DerivativeParsedMaterial
    f_name = degradation
    args = 'c effective_plastic_strain'
    function = '(1.0-c)^(2*effective_plastic_strain/critical_effective_plastic_strain)*(1.0 - eta) + eta'
    constant_names       = 'eta critical_effective_plastic_strain'
    constant_expressions = '1.0e-3 1.0e-2'
    derivative_order = 2
  [../]
```
For that, I defined the "effective_plastic_strain" in my .C UOBasedCP file and for crystal plasticity setup, it works perfectly 
```
_effective_plastic_strain(declareProperty<Real>("effective_plastic_strain")),
_effective_plastic_strain_old(
        getMaterialPropertyOld<Real>("effective_plastic_strain"))
```
and also can take it as output:
```
[./effective_plastic_strain]
    order = CONSTANT
    family = MONOMIAL
  [../]
[./effective_plastic_strain]
    type = MaterialRealAux
    variable = effective_plastic_strain
    property = effective_plastic_strain
    execute_on = timestep_end
  [../]
```
So, with these modifications, I tried to run my phase field code. I just wanted to confirm if this approach is correct?
The other approach I could think of was that I calculate the dDdc and d2Dd2c inside the .C files, but that will be too hectic, and there are no implementations like that that could be found.
I could not find much about the coupled variables that were implemented that way I did, even when looking in different .i files.

**Comment by GiudGiud** [comment]
Hello

Looks fine to me. @dschwen any thoughts on this phase field problem?

> I wanted to modify it as:

looks fine but the second one uses deprecated parameter names.
Function should be expression
f_name should be property_name. and so on
https://mooseframework.inl.gov/source/materials/DerivativeParsedMaterial.html
